website/src/components/feedback/index.js

```diff
@@ -0,0 +1,49 @@
+import React, { useState } from 'react';
+import { generateFeedback, sendToSegment } from '@site/src/components/feedback/feedbackHelper';
+
+const FeedbackWidget = () => {
+  const [feedback, setFeedback] = useState({
+    helpful: '',
+  });
+
+  const handleHelpfulChange = async (value) => {
+    // Send the feedback to Segment first
+    const feedbackJSON = generateFeedback(value);
+    await sendToSegment(feedbackJSON);
+
+    // Then open the Intercom widget if "No" is chosen
+    if (value === 'no') {
+      if (typeof Intercom !== 'undefined') {
+        Intercom('show');
+      }
+    }
+
+    // Update the feedback state after handling
+    setFeedback({
+      ...feedback,
+      helpful: value,
+    });
+  };
+
+  return (
+    <div className="feedback-widget-container">
```

Comment:
Using a `<form>` element would be semantic HTML. Will help with keyboard navigation, screen readers, etc.
Will have to change the handler and call it `onSubmit` for this to work. So leave it up to you if you want to change this now, or later if we add comments to the feedback as well.

Reply:
That's true. However, as there is no input we are capturing, and triggering segment as soon as the user chooses an option, I don't think we need a `<form/>` element here. Also, as the survey will cater to capturing detailed feedback comments, we are not adding comments to the feedback widget.